### PR TITLE
Add Info Bar when we can't find node

### PR DESCRIPTION
**Pull request by paulvanbrenk**
Show an info bar on project open/create instead of just a messagebox
on debug. This should help users getting started.

### CODE REVIEW

File: Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs [hunk 1]
```diff
@@ -46,6 +48,7 @@ public NodejsProjectNode(NodejsProjectPackage package) : base(package, null)
             InitNodejsProjectImages();
 #pragma warning restore 0612
 
+            infoBarManager = new MissingNodeInfoBar(package);
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs]: >package [](start = 52, length = 7)

Does the base type expose `package`?  If so, would it be preferable to create the `MissingNodeInfoBar` on demand?

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                var logicalView = VSConstants.LOGVIEWID_Primary;
+                if (ErrorHandler.Succeeded(projectNode.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)VsHierarchyPropID.ProjectDesignerEditor, out var editorType)) &&
+                    ErrorHandler.Succeeded(projectNode.OpenItemWithSpecific(VSConstants.VSITEMID_ROOT, 0, ref editorType, "", ref logicalView, (IntPtr)(-1), out var frame)))
+                {
+                    frame?.Show();
+                }
+            }
+        }
+
+        public void Close()
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: >public [](start = 8, length = 6)

private?

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            }
+
+            public void OnClosed(IVsInfoBarUIElement infoBarUIElement) => this.OnClose();
+
+            public void OnActionItemClicked(IVsInfoBarUIElement infoBarUIElement, IVsInfoBarActionItem actionItem)
+            {
+                if (actionItem.ActionContext is Action action)
+                {
+                    action.Invoke();
+                }
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: Alternatively, `(actionItem.ActionContext as Action).Invoke()`.

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs (hunk 2 above)
Comment (amcasey) [reply Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: Alternatively, consider inlining.

---
In reply to: [148164700](https://github.com/Microsoft/nodejstools/pull/1751#discussion_r148164700) [](ancestors = 148164700)

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            }
+
+            void ShowProjectProperties()
+            {
+                // open Project Properties
+                var logicalView = VSConstants.LOGVIEWID_Primary;
+                if (ErrorHandler.Succeeded(projectNode.GetGuidProperty(VSConstants.VSITEMID_ROOT, (int)VsHierarchyPropID.ProjectDesignerEditor, out var editorType)) &&
+                    ErrorHandler.Succeeded(projectNode.OpenItemWithSpecific(VSConstants.VSITEMID_ROOT, 0, ref editorType, "", ref logicalView, (IntPtr)(-1), out var frame)))
+                {
+                    frame?.Show();
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: >frame [](start = 20, length = 5)

Can this be null?

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs (hunk 2 above)
Comment (amcasey) [reply Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: Once it's inlined, there's no real reason to instantiate the type.  Show can just be a static method.

---
In reply to: [148165049](https://github.com/Microsoft/nodejstools/pull/1751#discussion_r148165049) [](ancestors = 148165049,148164700)

File: Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+            var actionItems = new[]
+            {
+                new InfoBarHyperlink(Resources.ConfigureProjectProperties, openProjectProps),
+                new InfoBarHyperlink(Resources.DownloadNodejs, downloadNode),
+            };
+
+            var infoBarModel = new InfoBarModel(Resources.NodejsNotInstalledInfoBar, actionItems, isCloseButtonVisible: true, image: KnownMonikers.StatusError);
+
+            uint eventCookie = 0;
```
Comment (amcasey) [inline Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: > = 0 [](start = 28, length = 4)

No reason to initialize this.

Comment (paulvanbrenk) [reply Nodejs/Product/Nodejs/Project/MissingNodeInfoBar.cs]: compiler disagrees.

**Pull request merged**